### PR TITLE
Spaces let users switch between Home and All Rooms behaviours

### PR DESCRIPTION
**Pull request by t3chguy**
Makes it so that a page reload is no longer required and takes them out of the feature/labs/beta namespace. It will mean the option is reset for users and the default has also changed.

Fixes https://github.com/vector-im/element-web/issues/18093

image
image
image

https://user-images.githubusercontent.com/2403652/127362571-a36fff09-839c-4649-8f41-a999ea2c2efc.mov

### CODE REVIEW

**Review by niquewoodhouse — CHANGES_REQUESTED** [review]
This is much better for the user than what I was originally thinking - a link to Preferences in settings.

Some small comments 

- Name of 'Space' should always be Home, whatever preferences are selected (the icon is a house, and that's just an option within the...house)
- Text could be 'show all rooms' (don't think you need the 'in Home', given the context
- I'm not personally convinced of the settings icon.  It might be because I personally think the context menu design puts too much emphasis on the icons to the left, but it just kind of sticks out as random? Or is that just me?  Any ideas of something better?  I was thinking the hash icon for 'rooms' might make more sense, but that's used in places already....so maybe it's not that good, but it would copy the "favourite" item on each room's context menu.

**Comment by t3chguy** [comment]
> Name of 'Space' should always be Home, whatever preferences are selected (the icon is a house, and that's just an option within the...house)

This should already be the case

> Text could be 'show all rooms' (don't think you need the 'in Home', given the context

Will change

> I was thinking the hash icon for 'rooms' might make more sense, but that's used in places already....so maybe it's not that good, but it would copy the "favourite" item on each room's context menu.

Will change

**Comment by nadonomy** [comment]
> I'm not personally convinced of the settings icon. It might be because I personally think the context menu design puts too much emphasis on the icons to the left, but it just kind of sticks out as random? Or is that just me? Any ideas of something better? I was thinking the hash icon for 'rooms' might make more sense, but that's used in places already....so maybe it's not that good, but it would copy the "favourite" item on each room's context menu.

@niquewoodhouse Agree! I wonder if we could reduce to no icon and just simply 'Show all rooms'?

**Comment by niquewoodhouse** [comment]
> @niquewoodhouse Agree! I wonder if we could reduce to no icon and just simply 'Show all rooms'?

I'm not sure if the component can handle having no icon - @t3chguy?

**Comment by t3chguy** [comment]
It is called the IconizedContextMenu but I can provide it an iconClass which does magic

image


Have just pushed the above

**Review by niquewoodhouse — APPROVED** [review]

File: src/settings/Settings.tsx [hunk 1]
```diff
@@ -758,6 +749,12 @@ export const SETTINGS: {[setting: string]: ISetting} = {
         supportedLevels: LEVELS_ACCOUNT_SETTINGS,
         default: null,
     },
+    "Spaces.all_rooms_in_home": {
```
Comment (turt2live) [inline src/settings/Settings.tsx:752]: the naming scheme for dotted settings is `Spaces.allRoomsInHome` - there used to be a style note somewhere in this file, but that seems to have gone missing.

**Review by turt2live — APPROVED** [review]
otherwise lgtm

**Pull request merged**